pkg/util/errors/k8sutil/k8sutil.go

```diff
@@ -0,0 +1,39 @@
+package k8sutil
+
+import (
+	"github.com/rancher/opni/pkg/util/errors"
+	"google.golang.org/grpc/codes"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func GRPCFromK8s(err error) *errors.GRPCError {
+	switch k8serrors.ReasonForError(err) {
+	case metav1.StatusReasonNotFound, metav1.StatusReasonGone:
+		return errors.New(err, codes.NotFound)
+	case metav1.StatusReasonUnauthorized:
+		return errors.New(err, codes.Unauthenticated)
+	case metav1.StatusReasonForbidden:
+		return errors.New(err, codes.PermissionDenied)
+	case metav1.StatusReasonAlreadyExists:
+		return errors.New(err, codes.AlreadyExists)
+	case metav1.StatusReasonInvalid:
+		return errors.New(err, codes.InvalidArgument)
+	case metav1.StatusReasonServerTimeout, metav1.StatusReasonTooManyRequests:
+		return errors.New(err, codes.ResourceExhausted)
+	case metav1.StatusReasonTimeout:
+		return errors.New(err, codes.DeadlineExceeded)
+	case metav1.StatusReasonBadRequest, metav1.StatusReasonMethodNotAllowed, metav1.StatusReasonConflict:
```

Comment:
From the codes documentation:
```//  (d) Use FailedPrecondition if the client performs conditional
	//      REST Get/Update/Delete on a resource and the resource on the
	//      server does not match the condition. E.g., conflicting
	//      read-modify-write on the same resource.
```
So I would argue Conflict aligns to FailedPrecondition

Method not allowed is a tricky one.  From the k8s documentation:
```
// StatusReasonMethodNotAllowed means that the action the client attempted to perform on the
	// resource was not supported by the code - for instance, attempting to delete a resource that
	// can only be created. API calls that return MethodNotAllowed can never succeed.
	// Status code 405
```
So it _could_ fall into the same logic for FailedPrecondition.  I don't think it aligns particularly well to unimplemented because it's more relating to a resource.  Maybe InvalidArgument could fit here?

For BadRequest the k8s docs state:
```
// StatusReasonBadRequest means that the request itself was invalid, because the request
	// doesn't make any sense, for example deleting a read-only object.  This is different than
	// StatusReasonInvalid above which indicates that the API call could possibly succeed, but the
	// data was invalid.  API calls that return BadRequest can never succeed.
	// Status code 400
```
I think this indicates quite a difference to InvalidArgument and falls more under the failed precondition.